### ISSUE TITLE
Add Cloud9 CLI support with 'Environment type' to 'Create and run in remote server SSH connection)'

### ISSUE THREAD

**Issue by ed-barberis** [aws/aws-cli#6985]
### Describe the feature

Add support to Cloud9 CLI with 'Environment type' to 'Create and run in remote server (SSH connection)' with an existing EC2 instance.

### Use Case

We use Cloud9 for customer enablement and all of the provisioning is automated to deploy an EC2 instance from a custom AMI w/ pre-installed C9 tooling and the Systems Manager Public Key, except for connecting/creating a Cloud9 environment associated with the running instance.

This is currently a manual step, but our goal is end-to-end automation.

### Proposed Solution

After preparing documentation on how-to prepare the AMI with the required Cloud9 artifacts, it seems that implementation of the CLI would be fairly straight forward.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CLI version used

2.7.2

### Environment details (OS name and version, etc.)

Amazon Linux 2

**Comment by tim-finnigan** [comment]
Hi @ed-barberis thanks for the feature request. Just to clarify, are you requesting for a new command to be added among these ones here: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloud9/index.html? If so could you elaborate a bit more on what you want the command to accomplish and how it would help your workflow?

**Comment by ed-barberis** [comment]
Yes. I would like a new command to be added that would perhaps be defined as '[CreateEnvironmentSSH](https://docs.aws.amazon.com/cloud9/latest/user-guide/create-environment-ssh.html)'. Note, that in the first sentence on the documentation page it states, "You can't create an SSH environment with code."

Our use case is that we want to automate the provisioning of the EC2 instance associated with the Cloud9 IDE. Instead of using the standard "bare-bones" OS on the VM, we would like to pre-install tooling needed for our customer enablement workshops. (Tools, such as 'docker', 'kubectl', 'helm', 'lab environment variables', etc.)

For our workshops, we create a custom AMI with the required tooling using HashiCorp Packer, including the Cloud9 binaries (using c9-install.sh) and public key. Next, we might spin-up 60 EC2 instances based on this custom AMI using Terraform. After that, we have to create and manually associate 60 Cloud9 environments with these running EC2 instances using the SSH environment type.

An alternative solution would be to enhance the existing '[CreateEnvironmentEC2](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloud9/create-environment-ec2.html)' AWS CLI command with the ability to add our own Pubic Key. That way we could provision the EC2 instance after creation using a tool such as Ansible.

Another way of stating the problem is that there is currently no way to automatically access the Cloud9 EC2 instances remotely. I realize that in the current CLI implementation, that is by design. However, if the AWS Console supports our use case, would it be possible to enhance the CLI to support that use case as well?

**Comment by tim-finnigan** [comment]
Thanks @ed-barberis for following up here. I have forwarded your request to the Cloud9 team for feedback as this change would need to happen at the API level. With API-related requests we generally recommend reaching out through [AWS Support](https://aws.amazon.com/contact-us/) for more direct escalation and tracking. But I will transfer this issue to our cross-SDK repository and let you know when we hear back from the Cloud9 team.
